template.tpl

```diff
@@ -0,0 +1,1704 @@
+    "simpleValueType": true,
+    "defaultValue": false
+  },
+  {
+    "type": "GROUP",
+    "name": "snowplowEventMapping",
+    "displayName": "Snowplow Event Mapping Options",
+    "groupStyle": "ZIPPY_CLOSED",
+    "subParams": [
+      {
+        "type": "GROUP",
+        "name": "atomicRules",
+        "displayName": "Snowplow Atomic Properties Rules",
+        "groupStyle": "ZIPPY_CLOSED",
+        "subParams": [
+          {
+            "type": "CHECKBOX",
+            "name": "includeAllAtomicEventProperties",
+            "checkboxText": "Include all Snowplow Atomic Event Properties",
+            "simpleValueType": true,
+            "defaultValue": false,
+            "alwaysInSummary": false,
+            "help": "Indicates if all Snowplow atomic event properties should be included in the request body. Note that some of the Snowplow atomic properties are already mapped to common event properties by the Snowplow Client."
+          },
+          {
+            "type": "TEXT",
+            "name": "atomicNest",
+            "displayName": "Nest atomic event properties under key (optional)",
+            "simpleValueType": true,
+            "enablingConditions": [
+              {
+                "paramName": "includeAllAtomicEventProperties",
+                "paramValue": true,
+                "type": "EQUALS"
+              }
+            ],
+            "help": "Specify a key under which the atomic properties of a Snowplow event will be nested. Alternatively, leaving it blank adds those properties in the request body without nesting."
+          }
+        ]
+      },
+      {
+        "type": "GROUP",
+        "name": "selfDescRules",
+        "displayName": "Snowplow Self-Describing Event Rules",
+        "groupStyle": "ZIPPY_OPEN",
+        "subParams": [
+          {
+            "type": "CHECKBOX",
+            "name": "includeSelfDescribingEvent",
+            "checkboxText": "Include Self Describing event",
+            "simpleValueType": true,
+            "defaultValue": true,
+            "help": "Indicates if Snowplow Self Describing event data should be included in the request body.",
+            "alwaysInSummary": true
+          },
+          {
+            "type": "TEXT",
+            "name": "selfDescNest",
+            "displayName": "Nest self-describing event data under key (optional)",
+            "simpleValueType": true,
+            "enablingConditions": [
+              {
+                "paramName": "includeSelfDescribingEvent",
+                "paramValue": true,
+                "type": "EQUALS"
+              }
+            ],
+            "help": "Specify a key under which the Self Describing event data will be nested. Alternatively, leaving it blank adds to the request body without nesting."
+          }
+        ]
+      },
+      {
+        "type": "GROUP",
+        "name": "entityRules",
+        "displayName": "Snowplow Event Context Rules",
```

Comment:
![Screenshot 2022-01-04 at 11 53 31](https://user-images.githubusercontent.com/8820009/148056082-9cfd5e3c-60d9-437d-ae4b-8bb649fda975.png)

It's currently impossible to create a tag that simply ignores all the context on an event. It forces you to include them all or at least pick one. I think the image in this screenshot should be valid?

Reply:
Thank you for reviewing @paulboocock ! Added the changes on a separate commit.
Indeed it was impossible to ignore all contexts, my oversight. The "non emtpy" validation was removed for the table. The check in code exists in [these lines](https://github.com/snowplow/snowplow-gtm-server-side-http-request-tag/blob/a01ff760da8f298e947a78ab16a7b7b76c4d5a6f/template.tpl#L741-L753), so no other change was needed.